### PR TITLE
Blog post - Hawkular APM's Vert.x OpenTracing example

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
+= Hawkular APM improvements for OpenShift
+Juraci Paixão Kröhling
+2016-12-16
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, apm, opentracing, vertx, openshift
+
+On our last release, we added a few improvements for applications deployed in
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/On our last release/In our last release/

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+OpenShift, such as an automatic way of recognizing the service's name. This
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/automatic way of recognizing the service's name/automatic way to recognize the name of the service/

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+is composed by inspecting the OpenShift's build name for the application,
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/This is composed by inspecting/The name consists of/

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+prepending it with the namespace to avoid clashes. For instance, if you have
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/prepending it with/prepended with/

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ifndef::env-github[]
+image::/img/blog/2016/2016-12-16-service-name.png[Service name from environment]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-12-16-service-name.png[Service name from environment]
+endif::[]
+
+We also improved our Vert.x OpenTracing, promoting it to the `examples`
+directory on the main Hawkular APM source code repository. We refactored it and
+prepared an Ansible playbook, so that it can be deployed in under a few minutes,
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/in under a few minutes/just a few minutes/very quickly?

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+On a Fedora 25 with Docker, OpenShift Origin Clients and Ansible, all that
+it's required is this:
+```bash
+ansible-playbook vertx-opentracing.yml
+```
+
+Once the playbook finishes running, you'll have an OpenShift cluster running
+with Hawkular APM on it and our Eclipse Vert.x example "Order Manager",
+instrumented using OpenTracing APIs. Refer to the https://github.com/hawkular/hawkular-apm/tree/master/examples/vertx-opentracing[README]
```
Comment (ppalaga) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: s/https://github.com/hawkular/hawkular-apm/tree/master/examples/vertx-opentracing[README]/https://github.com/hawkular/hawkular-apm/tree/master/examples/vertx-opentracing#opentracing---microservices-with-vertx[README]

**Review by ppalaga — COMMENTED** [review]
I proposed some rewording on a few places. Feel free to accept or deny. Nice work overall!

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+==================
+For a completely fresh install on Fedora 25, the whole environment can be
+created with the following commands:
+
+```bash
+sudo dnf install -y docker origin-clients ansible maven
+sudo sh -c "echo \"INSECURE_REGISTRY='--insecure-registry 172.30.0.0/16'\" >> /etc/sysconfig/docker"
+sudo systemctl start docker
+mkdir -p /opt/hawkular-apm/ ; cd /opt/hawkular-apm/
+mvn clean install -Dcheckstyle.skip=true -Dlicense.skip=true -Dxml.skip=true -DskipTests -Dnoui
```
Comment (objectiser) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: This seems to be in wrong order - should it cd /opt, then clone, cd hawkular-apm, then mvn clean install.

**Comment by objectiser** [comment]
@jpkrohling Good work - just a comment about the ordering of commands in the Tip at the end.

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc (hunk 7 above)
Comment (jpkrohling) [reply src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: Thanks for pointing this out! I think the wrong order is the `git clone` command which is coming after `mvn`. I changed it on the follow up commit.

**Comment by jpkrohling** [comment]
All changes done on the follow up commit. @ppalaga , @objectiser , could you please re-review? If that looks good, I'll squash the commits.

**Comment by objectiser** [comment]
@jpkrohling :+1:

**Comment by jotak** [comment]
@jpkrohling +1 :+1:

**Comment by ppalaga** [comment]
LGTM

**Comment by pavolloffay** [comment]
Nice blog, :+1:

File: src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+the first build of the service `my-project.order-manager`.
+
+ifndef::env-github[]
+image::/img/blog/2016/2016-12-16-service-name.png[Service name from environment]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-12-16-service-name.png[Service name from environment]
+endif::[]
+
+We also improved our Vert.x OpenTracing, promoting it to the `examples`
```
Comment (pavolloffay) [inline src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: Can we make `examples` pointing to GH?

Comment (jpkrohling) [reply src/main/jbake/content/blog/2016/12/16/hawkular-apm-vertx-opentracing-openshift.adoc]: Done

**Pull request merged**